src/Symfony/Component/HttpFoundation/File/UploadedFile.php

```diff
@@ -236,6 +236,39 @@ public function move($directory, $name = null)
         throw new FileException($this->getErrorMessage());
     }
 
+    public function moveAndRename(string $targetDirectory, string $fileNamePattern)
```

Comment:
we might also need to add to the phpdoc all the placeholders available for that argument.